exporters/tests/test_in_memory_metric.py

```diff
@@ -0,0 +1,175 @@
+#
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+
+from unittest import mock
+
+import pytest
+from prometheus_client import REGISTRY
+from prometheus_client.registry import Collector
+
+from webhook.models.pelorus_webhook import CommitTimePelorusPayload, PelorusPayload
+
+# from pelorus.config import load_and_log
+from webhook.store.in_memory_metric import (
+    PelorusGaugeMetricFamily,
+    _pelorus_metric_to_dict,
+    pelorus_metric_to_prometheus,
+)
+
+metric_labels = list(_pelorus_metric_to_dict(CommitTimePelorusPayload).values())
+
+in_memory_test_committime_metrics = PelorusGaugeMetricFamily(
+    "test_committime_metrics",
+    "Test timestamp",
+    labels=metric_labels,
+)
+
+
+class CustomCommitCollector(Collector):
+    def __init_subclass__(cls) -> None:
+        super().__init_subclass__()
+
+        # make sure __hash__ is something prometheus' registry can handle properly.
+        cls.__hash__ = lambda self: id(self)  # type: ignore
+
+    def collect(self) -> PelorusGaugeMetricFamily:
+        yield in_memory_test_committime_metrics
+
+
+class TestInMemoryMetric:
+    def setup_method(self):
+        self.custom_collector = CustomCommitCollector()
+        REGISTRY.register(self.custom_collector)
+
+    def teardown_method(self):
```

Comment:
This works? doesn't have to inherit from unittest.TestCase?

But I would remove this and use the solution I created in the pagerduty PR

Reply:
Yes, because we use **pytest** and this is `setup_method` from **pytest** and not `setUp` from **unittest**, same for the `teardown_method` which is not `tearDown` from **unittest**, you can invoke the test with some verbose info and redirect stdout from the test with such simple commands for the one file:

```shell
$ source .venv/bin/activate
$ cd exporters/tests
$ pytest -s -vvv test_in_memory_metric.py
```